src/metabase/api/embed/common.clj

```diff
@@ -125,9 +125,22 @@
   not automatically converted. Thus we must do it ourselves here to make sure things are done as we'd expect.
   Also, any param values that are blank strings should be parsed as nil, representing the absence of a value."
   [query-params]
-  (-> query-params
-      (update-keys keyword)
-      (update-vals (fn [v] (if (= v "") nil v)))))
+  (letfn [(maybe-read [v]
+            (if (string? v)
+              (cond
+                (#{"true" "false"} v) (parse-boolean v)
+                (str/blank? v)        nil
+                :else                 (or (parse-long v)
+                                          (parse-double v)
+                                          v))
+              v))]
+    (-> query-params
+        (update-keys keyword)
+        (update-vals (fn [v]
+                       (if (and (not (string? v))
+                                (seq v))
+                         (mapv maybe-read v)
+                         (maybe-read v)))))))
```

Comment:
I only really care about turning numbers and booleans into their expected types, anything else can be kept as strings.

query-params come from the URL, like this: `/api/embedding_endpoints/some_token?my_param=true&my_param2=asdf&my_param2=wasd`

So I want that to end up as:
```
{:my_param true
 :my_param2 ["asdf" "wasd"]}
 ```
 
 And this normalize takes care of that. There's a test added to show this behaviour as well.